tracer.go

```diff
@@ -185,7 +185,11 @@ func (t *tracerImpl) startSpanInternal(
 	tags opentracing.Tags,
 ) opentracing.Span {
 	sp.tracer = t
-	sp.event = t.options.NewSpanEventListener()
```

Comment:
it was forcing users to always provide an even listener if they were using `NewWithOptions()`. This interface does not feel like a required part of the tracer.